prometheus/promhttp/instrument_server.go

```diff
@@ -438,30 +435,30 @@ func (r *responseWriterDelegator) Write(b []byte) (int, error) {
 	return n, err
 }
 
-type fancyResponseWriterDelegator struct {
+type classicFancyResponseWriterDelegator struct {
```

Comment:
Err... naming is getting a bit out of control here.

You use just `pushDelegator` above. How about calling this `fancyDelegator` and the "super-fancy" one with push `fancyPushDelegator`.